docs/public-cloud/gitlab-setup.md

```diff
@@ -0,0 +1,225 @@
+# Gitlab Setup Guide for CAF Sample App
+
+This document outlines the steps necessary to set up a sample CAF (Cloud Adoption Framework) application in GitLab. 
+
+To summarize:
+
+0. Clone the [symphony](https://github.com/aztfmod/symphon) repo.
+  - For now use the following branch: [/davesee/bump_versions](https://github.com/aztfmod/symphony/tree/davesee/bump_versions)
+1. Set up the GitLab server with ```gitlab-server-setup.sh```
+2. Set up SSH Keys and clone repostories using ```clone-repos.sh```
+3. Deploy the launchpad landing zone manually to set up MSIs, using ```rover```
+4. Set up the Gitlab runners with ```install.sh```
+5. Run pipelines in Gitlab and troubleshoot as necessary
+
+# 1. GitLab Server Ssetup
+
+Set up the GitLab server using the instructions provided at the following location, specifically the section "Run the GitLab server deployment":
+
+* <a href="../../scripts/gitlab-server/README.md">/scripts/gitlab-server/README.md</a>
+
+# 2. Clone Repositories
+
+There are 2 options to clone the necessary repositories into the target location:
+* Option A (remote to remote): this requires Personal Access Tokens (PATs) for both the source and target repositories
+* Option B (local to remote): this requires a PAT only for the target repository
+
+TIPS: 
+- Create your Personal Access Token (PAT) on each GitLab location (source and/or target) and make a note of them.
+- Set up SSH Keys as needed.
+- Use Option B to get the latest current version of the repositories.
+
+For Option A (remote to remote), run the following command at the following location:
+
+```bash
+cd scripts/utils/
+
+./clone-repos.sh 
+-g reference_app_caf 
+-sp <source-pat> 
+-sd <source-fqdn> 
+-tp <target-pat> 
+-td <target-fqdn>
+-e <launchpad-environment>
+```
+
+The parameters are defined as follows:
+* g = GitLab Source Group
+* sp = (Source) Personal Access Token
+* sd = (Source) Fully Qualified Domain Name
+* tp = (Target) Personal Access Token
+* td = (Target) Fully Qualified Domain Name
+* e = CAF Launchpad Environment, e.g. "demo"
+
+For Option B (local to remote), run the following command at the following location:
+
+```bash
+cd scripts/utils/
+
+./clone-repos.sh 
+-s /workspaces/symphony/caf 
+-tp <target-pat>
+-td <target-fqdn>
+-e <launchpad-environment>
+-d  
+-o
+```
+
+The parameters are defined as follows:
+* s = Local Path for Source
+* tp = (Target) Personal Access Token
+* td = (Target) Fully Qualified Domain Name
+* e = CAF Launchpad Environment, e.g. "demo"
+* d = debug flag
+* o = overwrite target repo
+
+# 3. Launchpad Deployment
+
+To deploy the launchpad and create the necessary MSIs, 
+
+TIPS:
+* Get the latest version of the Azure CLI if necessary
```

Comment:
this is not necessary since the rover container is updated and has the latest cli.. granted that may become outdated at some point too!

Reply:
Good point, but I've decided to leave this tip in, since I have the "if necessary" qualifier, just in case.